doc/keymap: add documentation for xkb_symbols (#205)
xkbcommon/libxkbcommon: doc/keymap-format-text-v1.md

## The `xkb_symbols` section

NOTE: This section is incomplete.

This section is the fourth to be processed, after `xkb_keycodes`, `xkb_types`
and `xkb_compat`.

Statements of the form:

    xkb_symbols "basic" {
        ...
    }

> Declare a symbols map named `basic`. Statements inside the curly braces will
> affect only affect the symbols map.

Reviewer: ```suggestion
Declare a symbols map named `basic`. Statements inside the curly braces only
affect the symbols map.
```

## The `xkb_symbols` section

> 

Reviewer: Since the symbols section has a bunch more statements which aren't described here, I'd like to add the following disclaimer for now:

```suggestion

NOTE: This section is incomplete.

```

This section is the fourth to be processed, after `xkb_keycodes`, `xkb_types`
and `xkb_compat`.

Statements of the form:

    xkb_symbols "basic" {
        ...
    }

> Declares a symbols map named `basic`. Statements inside the curly braces will
> affect only affect the symbol map.

Reviewer: ```suggestion
Declare a symbols map named `basic`. Statements inside the curly braces will
only affect the symbols map.
```

> Options can be specified above the statement, separated by one space:

Reviewer: The code calls these "flags"; I prefer we use this term instead of "options" because there is already another concept called "options" and it's probably better to use distinct terms when possible to reduce the possibility of confusion.

Also I don't think it needs to be separated by exactly one space really.

```suggestion
A map can have various flags applied to it above the statement, separated by whitespace:
```

    partial alphanumeric_keys
    xkb_symbols "basic" {
        ...
    }

> The possible options are the following:

Reviewer: ```suggestion
The possible flags are the following:
```

  * `partial` - Indicates that the map doesn't cover a complete keyboard

>   * `default` - Marks the symbol map as the default variant when no variant is
>     specified

Reviewer: ```suggestion
  * `default` - Marks the symbol map as the default map in the file when no explicit
    map is specified. If no map is marked as a default, the first map in the file is the
    default.
```

  * `hidden` - Variant that can only be used internally
  * `alphanumeric_keys` - Indicates that the map contains alphanumeric keys
  * `modifier_keys` - Indicates that the map contains modifier keys
  * `keypad_keys` - Indicates that the map contains keypad keys
  * `function_keys` - Indicates that the map contains function keys
  * `alternate_group` - Indicates that the map contains keys for an alternate
    group

> If no `*_keys` options is supplied, then the map is assumed to cover a complete

Reviewer: ```suggestion
If no `*_keys` flags are supplied, then the map is assumed to cover a complete
```

keyboard.

> 

Reviewer: ```suggestion

At present, except for `default`, none of the flags affect key processing in
libxbkcommon, and only serve as metadata.

```

### Name statements

Statements of the form:

>     name[Group1] = "US/ASCII";

Reviewer: ```suggestion
    name[Group1] = "US/ASCII";
    groupName[1] = "US/ASCII";
```

Gives the name "US/ASCII" to the first group of symbols. Other groups can be
named using a different group index (ex: `Group2`), and with a different name.
A group must be named.

> 

Reviewer: ```suggestion

`group` and `groupName` mean the same thing, and the `Group` in `Group1` is optional.

```

### Include statements

Statements of the form:

    include "nokia_vndr/rx-51(nordic_base)

> Will include data from another `xkb_symbols` section, located in another file.

Reviewer: ```suggestion
Will include data from another `xkb_symbols` section, possibly located in another file.
```

Here it would include the `xkb_symbols` section called `nordic_base`, from the

> file `rx-51` located in the `nokia_vndr` folder, itself located in the
> `/usr/share/X11` folder.

Reviewer: ```suggestion
file `rx-51` located in the `nokia_vndr` folder, itself located in an
XKB include path.
```

### Key statement

Statements of the form:

    key <AD01> { [ q, Q ] };

Describes the mapping of a keycode `<AD01>` to a given group of symbols. The
symbols are `q` and `Q` in this example.

The possible keycodes are the keycodes defined in the `xkb_keycodes` section.

> Symbols are named using the symbolic names from the `keysymdef.h` file. A group

Reviewer: ```suggestion
Symbols are named using the symbolic names from the `xkbcommon/xkbcommon-keysyms.h` file. A group
```

of symbols is enclosed in brackets and separated by commas. Each element of the
symbol arrays corresponds to a different modifier level.

#### Groups

Each group represents a list of symbols mapped to a keycode:

    name[Group1]= "US/ASCII";
    name[Group2]= "Russian";
    ...
    key <AD01> { [ q, Q ],
                 [ Cyrillic_shorti, Cyrillic_SHORTI ] };

The long-form syntax can also be used to represent groups:

    key <AD01> {
        symbols[Group1]= [ q, Q ],
        symbols[Group2]= [ Cyrillic_shorti, Cyrillic_SHORTI ]
    };

> Groups can also be omitted, but the brackets must be present. The following
> statement will only defines the Group3 of a mapping

Reviewer: ```suggestion
Groups can also be omitted, but the brackets must be present. The following
statement only defines the Group3 of a mapping:
```
